### PR TITLE
[github-actions] retry coverage uploading on failures

### PR DESCRIPTION
**Pull request by Moandor-y**
Fixes <https://github.com/openthread/openthread/issues/5753>.

### CODE REVIEW

File: script/test [hunk 1]
```diff
@@ -420,14 +420,27 @@ do_generate_coverage()
...
     done
     lcov "${args[@]}" -o final.info
-    bash <(curl -s https://codecov.io/bash) -f final.info -Z
+
+    local retry_count=5
+    local fail_count=0
+    until ((fail_count > retry_count)); do
+        bash <(curl -s https://codecov.io/bash) -f final.info -Z && break
+        ((++fail_count))
+        echo >&2 "Upload failed, retrying (${fail_count}/${retry_count})"
```
Comment (bukepo) [inline script/test:438]: sleep for a while?

Comment (Moandor-y) [reply script/test:438]: Done.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**